cdip_connector/core/connector_base.py

```diff
@@ -23,30 +21,35 @@ class AbstractConnector(ABC):
     DEFAULT_REQUESTS_TIMEOUT = (3.1, 20)
 
     def __init__(self):
-        super().__init__()
+        # super().__init__()
+
+        logger = logging.getLogger(self.__class__.__name__)
+        logger.setLevel(cdip_settings.LOG_LEVEL)
+
         self.portal = PortalApi()
-        self.metrics = CdipMetrics()
+
         self.load_batch_size = 100  # a default meant to be overridden as needed
+        self.concurrency = 5
+
 
     def execute(self) -> None:
-        self.metrics.incr_count(MetricsEnum.INVOKED)
         asyncio.run(self.main())
 
     async def main(self) -> None:
         try:
             # Fudge with a really long timeout value.
             async with ClientSession(timeout=ClientTimeout(total=CLIENT_TIMEOUT_TOTAL)) as session:
-                logger.info(f'CLIENT_ID: {cdip_settings.KEYCLOAK_CLIENT_ID}')
-                integration_info = await self.portal.get_authorized_integrations(session)
+                self.logger.info(f'CLIENT_ID: {cdip_settings.KEYCLOAK_CLIENT_ID}')
+                integrations = await self.portal.get_authorized_integrations(session)
 
-                result = [await self.extract_load(session, i) for i in integration_info]
-                # tasks = [asyncio.ensure_future(self.extract_load(session, i)) for i in integration_info]
-                # result = await asyncio.gather(*tasks)
-                logger.info(result)
+                # result = [await self.extract_load(session, i) for i in integration_info]
+                for idx in range(0, len(integrations), self.concurrency):
+                    tasks = [asyncio.ensure_future(self.extract_load(session, integration)) for integration in integrations[idx:idx+self.concurrency]]
+                    result = await asyncio.gather(*tasks)
```

Comment:
I prefer using a Semaphore here, but Kubeless won't allow it. Once we're done with Kubeless we'll introduce a semaphore to make this quite a bit better.